### PR TITLE
Added cookie affinity filter

### PR DESCRIPTION
**Pull request by emmdurin**
To duplicate cookies from a path to another

This filter can be used by adding it to a route :
```
  filters:
  - name: CookieAffinity
    args:
      name: cookieName
      from: /originalPath
      to: /destinationPath
```

### CODE REVIEW

File: gateway/src/main/java/org/georchestra/gateway/filter/headers/CookieAffinityGatewayFilterFactory.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        public @Override Mono<Void> filter(ServerWebExchange exchange, GatewayFilterChain chain) {
+            return chain.filter(exchange).then(Mono.fromRunnable(() -> {
+                exchange.getResponse().getHeaders().getValuesAsList("Set-Cookie").stream()
+                        .flatMap(c -> java.net.HttpCookie.parse(c).stream())
+                        .filter(cookie -> cookie.getName().equals(config.getName())
+                                && cookie.getPath().equals(config.getFrom()))
+                        .forEach(cookie -> {
+                            ResponseCookie responseCookie = ResponseCookie.from(cookie.getName(), cookie.getValue())
+                                    .domain(cookie.getDomain()).httpOnly(cookie.isHttpOnly()).secure(cookie.getSecure())
+                                    .maxAge(cookie.getMaxAge()).path(config.getTo()).build();
```
Comment (jahow) [inline gateway/src/main/java/org/georchestra/gateway/filter/headers/CookieAffinityGatewayFilterFactory.java:56]: Maybe a stupid remark but isn't there a way to just duplicate the cookie and change the path? I think that's how it was done in the SP but the cookie class used was also different.

**Comment by edevosc2c** [comment]
need reformat

File: gateway/src/main/java/org/georchestra/gateway/filter/headers/CookieAffinityGatewayFilterFactory.java (hunk 1 above)
Comment (emmdurin) [reply gateway/src/main/java/org/georchestra/gateway/filter/headers/CookieAffinityGatewayFilterFactory.java:56]: Security proxy has also a code like this which assigns fields one by one (see below). The problem is that there is no constructor or builder that can take another `ResponseCookie`. Also note that the `ResponseCookie` class in Security proxy is not the same as the one in Gateway, they are located in different namespaces and are not created the same way.

Code from Security Proxy :
```
        Cookie responseCookie = new Cookie(cookie.getName(), cookie.getValue());
        if (null != cookie.getDomain())
            responseCookie.setDomain(cookie.getDomain());
        responseCookie.setHttpOnly(responseCookie.isHttpOnly());
        responseCookie.setSecure(cookie.getSecure());
        responseCookie.setVersion(cookie.getVersion());
        responseCookie.setMaxAge(responseCookie.getMaxAge());
        responseCookie.setComment(cookie.getComment());
        responseCookie.setPath(targetPath);
```

**Comment by emmdurin** [comment]
> need reformat

Yes, and cleaning of remaining commented code.

**Review by jahow — APPROVED** [review]
Not tested but everything looks in order to me, thanks!

**Pull request merged**